fabric-item-api-v1/src/main/java/net/fabricmc/fabric/mixin/item/AbstractFurnaceBlockEntityMixin.java

```diff
@@ -41,8 +42,11 @@ private static void getStackCraftingRemainder(World world, BlockPos pos, BlockSt
 		capturedInventory = ((AbstractFurnaceBlockEntityMixin) (Object) blockEntity).inventory;
 	}
 
-	@Redirect(method = "tick", at = @At(value = "INVOKE", target = "Lnet/minecraft/item/ItemStack;isEmpty()Z", ordinal = 2))
-	private static boolean setStackCraftingRemainder(ItemStack stack) {
+	@Redirect(method = "tick", at = @At(value = "INVOKE", target = "Lnet/minecraft/item/ItemStack;isEmpty()Z"), slice = @Slice(from = @At(value = "INVOKE", target = "Lnet/minecraft/item/ItemStack;decrement(I)V")))
+	private static boolean setStackCraftingRemainder(ItemStack oldStack) {
+		// We need the item stack before it gets decremented.
+		ItemStack stack = capturedInventory.get(1);
```

Comment:
Will this really work? Since it's the same stack instance as the modified one, it'll already have been decremented as far as I can tell.

Reply:
Some testing seems to show it works fine.

Reply:
It doesn't - the stack is decremented ⇒ it becomes empty ⇒ `getItem` returns `minecraft:air` ⇒ `FabricItemStack.getRecipeRemainder` returns `ItemStack.EMPTY`.

https://user-images.githubusercontent.com/6596629/188511189-1073029d-47f5-4f3a-b27b-886a38c04b9c.mp4